antsibull/docs_parsing/__init__.py

```diff
@@ -62,3 +62,33 @@ def _get_environment(collection_dir: t.Optional[str]) -> t.Dict[str, str]:
             if env_var in os.environ:
                 env[env_var] = os.environ[env_var]
     return env
+
+
+class AnsibleCollectionInfo:
+    path: str
+    version: t.Optional[str]
+
+    def __init__(self, path: str, version: t.Optional[str]):
+        self.path = path
+        self.version = version
+
+    def __repr__(self):
+        return 'AnsibleCollectionInfo({0}, {1})'.format(repr(self.path), repr(self.version))
+
+
+class AnsibleCollectionDocs:
```

Comment:
It is essentially a `NamedTuple`. I've converted it into a 'proper' `NamedTuple` in 0996b65bc9c37f45cd7269266cda2cef022b6ce7. I don't think it is easier to read though.

Reply:
Yeah.... what I mean is that the two things are really items which are unassociated with each other, so the API should look like:
``` python

def get_disparate_information(url):
   return len(url), contents_stored_at(url)

length, contents = get_disparate_information('https://google.com/')

do_something_else(length)
yet_another_else(contents)
```

Packing them together, at least as arbitrary concatenation, should be temporary (so that we can return two values from where the parsing happened) but the API shouldn't encourage people to store and access the values as a single entity.

Reply:
Is 892998d49d2f0d32bccd07b7e5831d131e0d22c4 ok?